checkstyle/checkstyle.xml

```diff
@@ -73,7 +73,7 @@
     <!-- Checks whether files end with a new line.                        -->
     <!-- See http://checkstyle.sf.net/config_misc.html#NewlineAtEndOfFile -->
     <module name="NewlineAtEndOfFile">
-        <property name="lineSeparator" value="lf"/>
+        <property name="lineSeparator" value="lf_cr_crlf"/>
```

Comment:
All our code uses unix endlines. You probably need to configure your GIT to not change line endings when checking out. I'm on Windows too.

Reply:
Ohh, that should be the way to go. thanks :)

Reply:
I am thinking about the linked issue #18727 and it seems we could either introduce `git config —global core.autocrlf false` command to our Jenkins build or a `.gitattributes` file to the Hazelcast project. Which one do you think is better?

Reply:
So, as you mentioned, the problem is, on Windows when you clone a remote repository it will automatically convert line endings to the version of Windows one (`\n` -> `\r\n`)

Reply:
I think we should base our decision on the defaults. If the default is `auto-crlf=true`, then we should go ahead with this PR. Otherwise, we’d basically be forcing all windows users to reconfigure git once checking out our PR, after they spend time figuring out what exactly is going wrong.

Reply:
[It seems](https://git-scm.com/book/tr/v2/Customizing-Git-Git-Configuration) `autocrlf = false` is the default on Git. However, Git installer on Windows sets this value to `true`, [unless chosen otherwise](https://stackoverflow.com/a/49207740) during the installation process. Since this is git related issue, maybe we should introduce `.gitattributes` file to enforce either the defaults or override some behavior if needed in the future. 

> Optionally, you can configure a .gitattributes file to manage how Git reads line endings in a specific repository. When you commit this file to a repository, it overrides the core.autocrlf setting for all repository contributors. This ensures consistent behavior for all users, regardless of their Git settings and environment.

-- [GitHub suggestion](https://docs.github.com/en/github/getting-started-with-github/getting-started-with-git/configuring-git-to-handle-line-endings)

Reply:
To me this sounds like the best option, nice finding @fbarotov. Personally, I'd keep the checkstyle rule for "lf" because ultimately that's the only option that what we want in the codebase. So, if `.gitattributes` is the "kosher" way to go here, let's do it.

Reply:
I've set the option `auto_crlf` to false on windows, i think it's best to have the same files bit by bit locally as they are in the repo. This option is legacy I'd say, today even Windows Notepad can open files with unix line endings, and even if you write, it produces them.

Reply:
But adding to the repo if we can is ideal i guess.

Reply:
thanks for the comments. `.gitattributes` is here now :)